core/src/main/java/io/aiven/klaw/service/EnvsClustersTenantsControllerService.java

```diff
@@ -898,6 +909,80 @@ public ApiResponse deleteEnvironment(String envId, String envType) throws KlawEx
     }
   }
 
+  private void removeEnvironmentMapping(String id, int tenantId, String envType)
+      throws KlawException {
+
+    if (KafkaClustersType.KAFKA.value.equals(envType)
+        || KafkaClustersType.SCHEMA_REGISTRY.value.equals(envType)) {
+      Env env = manageDatabase.getHandleDbRequests().selectEnvDetails(id, tenantId);
+      if (env.getAssociatedEnv() != null) {
+        Env linkedEnv =
+            manageDatabase
+                .getHandleDbRequests()
+                .selectEnvDetails(env.getAssociatedEnv().getId(), tenantId);
+        linkedEnv.setAssociatedEnv(null);
+        manageDatabase.getHandleDbRequests().addNewEnv(linkedEnv);
+      }
+    }
+  }
+
+  private EnvTag addEnvironmentMapping(
+      EnvTag tag, String id, String name, int tenantId, String envType)
+      throws KlawValidationException {
+    // only assignable on a schema registry
+    if (KafkaClustersType.SCHEMA_REGISTRY.value.equals(envType)) {
+      //      EnvTag existingTag = getKafkaAssociation(null, id, tenantId);
+      if (tag != null) {
+
+        associateWithKafkaEnv(tag, id, name, tenantId);
+        // remove existing association if it exists
+        removeAssociationWithKafkaEnv(id, tenantId);
```

Comment:
Is this required? as, it will be overwritten anyways from https://github.com/aiven/klaw/pull/697/files#diff-f7d922dcecfea3bdd2921bdb72ff1526ca7a213c1549e322e252657d04373ae4R714

Reply:
Yes, when switching in the schema view from say 'DEV' to 'TST' as the associated Kafka Env the previous Kafka Env needs to be updated here to remove that association which is what this piece here does.